File: SuBilet-frontend/src/services/CustomerService.js
import apiClient from '../utils/apiClient';

const API_URL = '/customers';

class CustomerService {
    
    getAllCustomers() {
        return apiClient.get(API_URL);
    }

    getCustomerById(id) {
        return apiClient.get(`${API_URL}/${id}`);
    }

    getCustomerByTcNo(tcNo) {
        return apiClient.get(`${API_URL}/tc/${tcNo}`);
    }

    createCustomer(customer) {
        return apiClient.post(API_URL, customer);
    }

    updateCustomer(id, customer) {
        return apiClient.put(`${API_URL}/${id}`, customer);
    }

    deleteCustomer(id) {
        return apiClient.delete(`${API_URL}/${id}`);
    }

    login(tcNo) {
        return apiClient.post(`${API_URL}/login`, { tcNo });
    }
}

const customerServiceInstance = new CustomerService();
export default customerServiceInstance;

File: SuBilet-backend/src/main/java/com/subilet/backend/service/CustomerService.java
package com.subilet.backend.service;

import com.subilet.backend.entity.Customer;
import com.subilet.backend.exception.ResourceNotFoundException;
import com.subilet.backend.repository.CustomerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CustomerService {

    @Autowired
    private CustomerRepository customerRepository;

    public List<Customer> getAllCustomers() {
        return customerRepository.findAll();
    }

    public Optional<Customer> getCustomerById(Integer id) {
        return customerRepository.findById(id);
    }

    public Customer createCustomer(Customer customer) {
        // TC No kontrolü - varsa mevcut müşteriyi döndür
        if (customer.getTcNo() != null && !customer.getTcNo().isEmpty()) {
            Optional<Customer> existing = customerRepository.findByTcNo(customer.getTcNo());
            if (existing.isPresent()) {
                return existing.get(); // Mevcut müşteriyi döndür
            }
        }
        
        // Username ve password kontrolü
        if (customer.getUsername() == null || customer.getUsername().isEmpty()) {
            throw new RuntimeException("Kullanıcı adı zorunludur");
        }
        if (customer.getPassword() == null || customer.getPassword().isEmpty()) {
            throw new RuntimeException("Şifre zorunludur");
        }
        
        return customerRepository.save(customer);
    }

    public Customer updateCustomer(Integer id, Customer customerDetails) {
        Customer customer = customerRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Müşteri bulunamadı: " + id));

        customer.setIsimSoyad(customerDetails.getIsimSoyad());
        customer.setDogumTarihi(customerDetails.getDogumTarihi());
        customer.setUyruk(customerDetails.getUyruk());
        customer.setCinsiyet(customerDetails.getCinsiyet());
        customer.setMail(customerDetails.getMail());
        customer.setTelNo(customerDetails.getTelNo());

        return customerRepository.save(customer);
    }

    public void deleteCustomer(Integer id) {
        customerRepository.deleteById(id);
    }

    public Optional<Customer> findByTcNo(String tcNo) {
        return customerRepository.findByTcNo(tcNo);
    }
}

